cdap-common/src/main/java/co/cask/cdap/common/lang/ProgramResources.java

```diff
@@ -115,8 +115,8 @@ public String apply(ClassPath.ClassInfo input) {
     Set<String> resources = getBaseResources();
 
     // Base on the type, add extra resources
-    // Current only Spark type has extra visible resources
-    if (type == ProgramType.SPARK) {
+    // Current only Spark and Workflow type has extra visible resources
+    if (type == ProgramType.SPARK || type == ProgramType.WORKFLOW) {
```

Comment:
Is it because Workflow can have Spark in them which will have extra visible resource ?
Nice!